packages/desktop-client/src/components/modals/KeyboardShortcuts.tsx

```diff
@@ -0,0 +1,241 @@
+import * as Platform from 'loot-core/src/client/platform';
+
+import { Modal, type ModalProps } from '../common/Modal';
+import { Text } from '../common/Text';
+import { View } from '../common/View';
+
+type KeyboardShortcutsModalProps = {
+  onAccounts: boolean;
+  modalProps?: Partial<ModalProps>;
+};
+
+type KeyIconProps = {
+  shortcut: string;
+};
+
+type GroupHeadingProps = {
+  group: string;
+};
+
+type ShortcutProps = {
+  shortcut: string;
+  description: string;
+  meta?: string;
+};
+
+function KeyIcon({ shortcut }: KeyIconProps) {
+  return (
+    <div
+      style={{
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        fontWeight: 'bold',
+        backgroundColor: '#fff',
+        color: '#000',
+        border: '1px solid #000',
+        borderRadius: 8,
+        minWidth: 35,
+        minHeight: 35,
+        filter: 'drop-shadow(1px 1px)',
+        padding: 5,
+      }}
+    >
+      {shortcut}
+    </div>
+  );
+}
+
+function GroupHeading({ group }: GroupHeadingProps) {
+  return (
+    <Text
+      style={{
+        fontWeight: 'bold',
+        fontSize: 16,
+        marginTop: 20,
+        marginBottom: 10,
+      }}
+    >
+      {group}:
+    </Text>
+  );
+}
+
+function Shortcut({ shortcut, description, meta }: ShortcutProps) {
+  return (
+    <div
+      style={{
+        display: 'flex',
+        marginBottom: 10,
+        marginLeft: 20,
+      }}
+    >
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+        }}
+      >
+        <div
+          style={{
+            display: 'flex',
+            flexDirection: 'row',
+            marginRight: 10,
+          }}
+        >
+          {meta && (
+            <>
+              <KeyIcon shortcut={meta} />
+              <Text
+                style={{
+                  display: 'flex',
+                  alignItems: 'center',
+                  textAlign: 'center',
+                  fontSize: 16,
+                  paddingLeft: 2,
+                  paddingRight: 2,
+                }}
+              >
+                +
+              </Text>
+            </>
+          )}
+          <KeyIcon shortcut={shortcut} />
+        </div>
+        <div
+          style={{
+            display: 'flex',
+            flexDirection: 'row',
+            flex: 1,
+          }}
+        />
+      </div>
+      <div
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          maxWidth: 300,
+        }}
+      >
+        {description}
+      </div>
+    </div>
+  );
+}
+
+export function KeyboardShortcuts({
+  onAccounts,
+  modalProps,
+}: KeyboardShortcutsModalProps) {
+  const ctrl = Platform.OS === 'mac' ? '⌘' : 'Ctrl';
```

Comment:
Maybe instead of updating the reducer to add a special case, we can check the modal stack here? If the last modal in the stack is this modal, pop the modal stack before returning.

Reply:
That would only handle stopping it from showing twice.  It would not handle the case of not showing the first time if another modal is already shown.

Reply:
Why shouldn't we allow it when other modals are open? I think we should leave the possibility for modals to have hotkeys modal too in case we need to add for one in the future e.g. add hotkeys for schedules modal

Reply:
I don't think modals will have their own hotkeys.  They might provide shortcuts to jump to a control, but if they do (and none do currently) those should be visibly displayed in the modal itself so it is obvious.